### DISCUSSION TITLE
Code Owners?

### DISCUSSION THREAD

**Discussion by ryanmrichard** [NWChemEx/nwchemex.github.io#3] [Ideas]
What do people think about establishing code owners for the stack?

To clarify, a code owner is someone who takes it upon themselves to be an expert in a repository. This entails fully understanding the design, vision, and existing source code in that repository. The reasons for having code owners is:

1. Establish point of contacts for questions/concerns related to a repository.
2. Have default reviewers for specific repositories.
3. Spread the code maintenance burden around.

If people are amenable to having code owners, we can use this thread to get volunteers. To be clear this post is largely targeting repos that live in the NWChemEx organization. As a reminder the 1.0 stack will include the following repos (which live in the NWChemEx organization):

- ParallelZone
- PluginPlay
- Chemist
- SimDE
- ChemCache
- TBD 
- FriendZone
- NWChemEx

The "TBD" repo is a repo that will contain a plugin wrapping the TAMM powered SCF and coupled cluster methods; IIRC GauXC is either integrated with TAMM SCF already or will be, meaning DFT will be there too. It's TBD because I'm not sure if we established if this will be a gutted version of the existing SCF repo, a variation on the CoupledCluster repo, or some new repo for this purpose.

**Comment by jwaldrop107** [comment]
Code ownership seems like a good idea to me. I've spent enough time in most of the repos, I can probably volunteer any of them.

**Comment by yzhang-23** [comment]
I also think assigning someone(s) to each repo makes the development cycle more efficient. However, just my personal feeling: maybe we can change the name "owner" to "manager" (or something similar), as no one actually owns a repo?

**Comment by jwaldrop107** [comment]
@yzhang-23 "code owner" is a fairly standard term and the one [Github uses](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners).

**Reply by yzhang-23** [reply]
Ok, I see. Thanks for pointing this out to me!
